src/linear_algebra/general_linear_group.lean

```diff
@@ -0,0 +1,269 @@
+end
+
+
+noncomputable instance : group (GL n R) :=
+{ mul := λ u₁ u₂, ⟨u₁.val * u₂.val, u₂.inv * u₁.inv,
+    by rw [mul_assoc, ← mul_assoc u₂.val, units.val_inv, one_mul, units.val_inv],
+    by rw [mul_assoc, ← mul_assoc u₁.inv, units.inv_val, one_mul, units.inv_val]⟩,
+  one := ⟨1, 1, one_mul 1, one_mul 1⟩,
+  mul_one := λ u,  mul_one u,
+  one_mul := λ u,  one_mul u,
+  mul_assoc := λ u₁ u₂ u₃,  mul_assoc u₁ u₂ u₃,
+  inv := general_linear_group.has_inv.inv,
+  mul_left_inv := λ u,  by {apply is_left_inv,} }
+
+-/
+ instance : group (GL n R) :=infer_instance
+
+/-- Applying `matrix.nonsing_inv` to the coercion is the same as coercing the group inverse. -/
+lemma nonsing_inv_coe_eq_coe_inv (A : GL n R) : (A : matrix n n R)⁻¹ = ↑(A⁻¹) :=
+begin
+  letI := A.invertible,
+  exact (inv_eq_nonsing_inv_of_invertible (A : matrix n n R)).symm,
+end
+
+
+instance SL_to_GL : has_coe (special_linear_group n R) (GL n R):=
+⟨λ A, ⟨↑A, ↑(A⁻¹), congr_arg coe (mul_right_inv A), congr_arg coe (mul_left_inv A)⟩⟩
+
+
+
+@[simp] lemma GL_vals (A : GL n R): ∀ i j, A i j = A.1 i j:=
+begin
+unfold_coes, intros i j,
+refl,
+end
+
+
+
+lemma det_not_zero [nontrivial R] (A: GL n R): det A ≠ 0:=
+begin
+have:=GL.is_unit_det _ _ A, simp, by_contradiction, unfold_coes at *,
+rw h at this, simp at *, exact this,
+ end
+
+
+
+end general_linear_group
+
+namespace GL_plus
+
+variables  (n : Type u) [decidable_eq n] [fintype n] (R : Type v)
+ [linear_ordered_comm_ring R ]
+
+
+lemma one_in_GL_pos : 0 < det (1:GL n R ) :=
+begin
+simp only [det_one, gt_iff_lt, general_linear_group.one_apply,zero_lt_one],
+end
+
+
+lemma mul_in_GL_pos (A B :GL n R ) (h1: 0<  det A  ) (h2: 0<  det B ): 0 <det (A*B):=
+
+begin
+simp only [gt_iff_lt, det_mul, mul_eq_mul], apply mul_pos h1 h2,
+end
+
+
+
+lemma inv_det_pos  (A:GL n R ) (h: 0 < det A ):  0 < det (A⁻¹).1 :=
+
+begin
+have h0:=(GL.is_unit_det _ _ A),
+have h1:=is_unit_nonsing_inv_det A h0,
+have h2:= nonsing_inv_det A h0,
+have h3: 0 < det ⇑A*  det (⇑A)⁻¹ , by {rw mul_comm, rw h2, linarith}, unfold_coes at *,
+convert (zero_lt_mul_left h).mp h3,  sorry,
+end
+
+
+
+/-- This is the subgroup of `nxn` matrices with  entries over a
+linear ordered ring and positive determinant-/
+
+
+noncomputable def GL_pos : subgroup (GL n R) :=
+{carrier:={M  | 0 < det M},
+ one_mem':=one_in_GL_pos _ _ ,
+ mul_mem':=λ A B h1 h2, mul_in_GL_pos _ _ A B h1 h2,
+ inv_mem':=λ A h1, by {have:= inv_det_pos _ _ A h1, simp only [set.mem_set_of_eq] at *,
+ unfold_coes at *, convert this, }
+}
+
+
+
+
+@[simp] lemma mem_GL_pos (A: GL n R ) :
+  A  ∈ (GL_pos n R)  ↔ 0 < A.1.det := iff.rfl
+
+
+instance GL_pos_to_GL : has_coe (GL_pos n R)  (GL n R) := ⟨λ A, A.val⟩
+
+
+
+lemma SL_det_pos' (A : special_linear_group n R): 0< (A).1.det:=
+
+begin
+have:=A.2, simp only [gt_iff_lt, subtype.val_eq_coe],
+ simp only [subtype.val_eq_coe] at this,  rw this,linarith,
+end
+
+noncomputable instance SL_to_GL_pos: has_coe (special_linear_group n R) (GL_pos n R):=
```

Comment:
Lean complains that this is computable after all now that you removed your group instance:
```suggestion
instance SL_to_GL_pos: has_coe (special_linear_group n R) (GL_pos n R):=
```

Reply:
done